homeassistant/components/here_weather/weather.py

```diff
@@ -0,0 +1,222 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Add here_weather entities from a config_entry."""
+    here_weather_data_dict = hass.data[DOMAIN][config_entry.entry_id]
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type != MODE_ASTRONOMY:
+            async_add_entities(
+                [
+                    HEREDestinationWeather(
+                        config_entry,
+                        here_weather_data_dict[sensor_type].coordinator,
+                        sensor_type,
+                    )
+                ],
+                True,
+            )
+
+
+class HEREDestinationWeather(CoordinatorEntity, WeatherEntity):
+    """Implementation of an HERE Destination Weather WeatherEntity."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, coordinator: DataUpdateCoordinator, mode: str
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._name = config_entry.data[CONF_NAME]
+        self._mode = mode
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} {self._mode}"
+
+    @property
+    def unique_id(self):
+        """Set unique_id for sensor."""
+        return f"{self._name}_{self._mode}"
+
+    @property
+    def condition(self):
+        """Return the current condition."""
+        return get_condition_from_here_data(self.coordinator.data)
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return get_temperature_from_here_data(self.coordinator.data, self._mode)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return convert_unit_of_measurement_if_needed(
+            self.coordinator.hass.config.units.name, TEMP_CELSIUS
+        )
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return None
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        get_attribute_from_here_data(self.coordinator.data, "humidity")
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        get_attribute_from_here_data(self.coordinator.data, "windSpeed")
+
+    @property
+    def wind_bearing(self):
+        """Return the wind bearing."""
+        get_attribute_from_here_data(self.coordinator.data, "windDirection")
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return None
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        if self.coordinator.data is None:
+            return None
+        data = []
+        for offset in range(len(self.coordinator.data)):
+            data.append(
+                {
+                    ATTR_FORECAST_TIME: get_attribute_from_here_data(
```

Comment:
Does this return an isoformatted datetime string?

Reply:
Yes in the form of `2019-11-19T00:00:00.000-05:00` I added the correct type in this commit: https://github.com/home-assistant/core/pull/28910/commits/15e174d00a013cf285c8f23eb018e079d3ca5e09